native/core/src/execution/jni_api.rs

```diff
@@ -346,6 +347,10 @@ pub unsafe extern "system" fn Java_org_apache_comet_Native_executePlan(
 
         let exec_context_id = exec_context.id;
 
+        // We maintain a map of metrics name -> jstring object to reduce the overhead of calling
+        // jni_NewStringUTF repeatedly.
+        let mut metrics_jstrings: HashMap<String, Arc<GlobalRef>> = HashMap::new();
```

Comment:
Rather than creating this hash map on each call to `executePlan`, could we create it during `createPlan` and store it in the `ExecutionContext`? This would allow us to allocate the strings once per plan rather than once per output batch. The resources would then be freed in `releasePlan`.

Reply:
Yep I think I can do that. I originally went down the path and tried to populate it during `CreatePlan` and then realized that I could only allocate the strings lazily. However, there's nothing stopping me from hoisting the data structure itself to the `ExecutionContext`. I do wonder about thread safety though.